src/main/java/seedu/address/logic/commands/addcommands/AddTaskToGroupCommand.java

```diff
@@ -32,58 +35,64 @@ public class AddTaskToGroupCommand extends Command {
         + "Parameters: "
         + PREFIX_TASK_NAME + "TASK_NAME "
         + PREFIX_TASK_DEADLINE + "TASK_DATE "
-        + PREFIX_GROUP_NAME + "GROUP_NAME\n"
+        + PREFIX_GROUP_NAME + "GROUP_NAME "
+        + "[" + PREFIX_GROUP_NAME + "GROUP_NAME]...\n"
         + "Example: " + COMMAND_WORD + " "
         + PREFIX_TASK_NAME + "Complete this task "
         + PREFIX_TASK_DEADLINE + "2024-01-01 1300 "
         + PREFIX_GROUP_NAME + "Team 1";
 
-    public static final String MESSAGE_SUCCESS = "Added task: %1$s to %2$s";
+    public static final String MESSAGE_SUCCESS = "Added task (%1$s) to the following groups:\n%2$s";
 
     public static final String MESSAGE_DUPLICATE_TASK_IN_GROUP = "This task is already in the group";
 
     private final TaskName taskName;
 
     private final Deadline deadline;
 
-    private final GroupName toAddInto;
+    private final Set<GroupName> toAddInto;
 
     /**
      * Creates an AddStudentToGroupCommand to add the specified {@code Task} to the specified {@code Group}.
      */
-    public AddTaskToGroupCommand(TaskName taskName, Deadline deadline, GroupName groupName) {
+    public AddTaskToGroupCommand(TaskName taskName, Deadline deadline, Set<GroupName> groupNames) {
         requireNonNull(taskName);
-        requireNonNull(groupName);
+        requireNonNull(groupNames);
         this.taskName = taskName;
         this.deadline = deadline;
-        this.toAddInto = groupName;
+        this.toAddInto = groupNames;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        if (!model.containsGroupName(toAddInto)) {
-            throw new CommandException(MESSAGE_GROUP_NAME_NOT_FOUND);
-        }
-
         Task task = new Task(taskName, deadline);
-        Group group = model.getGroupByName(toAddInto);
-
-        if (model.hasTaskInGroup(task, group)) {
-            throw new CommandException(MESSAGE_DUPLICATE_TASK_IN_GROUP);
-        }
-
-        model.addTaskToGroup(task, group);
-        if (!model.hasTask(task)) {
-            model.addTask(task);
-        } else {
-            model.increaseGroupWithTask(task);
+        List<GroupName> groups = toAddInto.stream().toList();
```

Comment:
Since you've added in the add existing task to group function is it better if we raise an exception if a user adds an existing task using this function?

Reply:
Do you mean here in AddTaskToGroup or AddExistingTaskToGroup?

I did miss out checking whether the task exists globally already though, will add that in

Reply:
wait actually does `model.increaseGroupWithTask` handle incrementing the counter if the task exists globally?

Reply:
I meant like if the task exists globally then if the user tries to add a task using this function, with the same task name and deadline as a task in the global list then we should disallow it because there's another function for it.

And yeah the increaseGroupWithTask function does increment the counter of the global task. The groupsWithTask parameter for a Task in a group does not matter. Because its function is to signal when a task is to be deleted from the global list.